include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,612 @@
+    {
+    }
+
+    void parse(argument_parser_meta_data const & parser_meta) 
+    {
+        rxml::xml_document<> *ctd_document = new rxml::xml_document<>();
+
+        // Validate the application name using the regex provided by the CTD XML schema.
+        if (!std::regex_match(parser_meta.app_name, 
+                              std::regex("[A-Za-z0-9?_\\-]+"))) {
+            throw parser_design_error("CTD file requires the application name to match [A-Za-z0-9_\\-]+ regex");
+        }
+
+        // Append the XML declaration node on top of the XML document. 
+        append_declaration_node(ctd_document,
+                                ctd_document);
+
+        // Append the tool subtree to the CTD document DOM tree.
+        append_tool_node(ctd_document, 
+                         ctd_document,
+                         parser_meta);
+        
+        // Print the CTD file on the standard output stream.
+        std::cout << *ctd_document;
+
+        std::exit(EXIT_SUCCESS);
+    }
+
+    /*!\brief Adds an help page section to the seqan3::argument_parser.
+     * \param[in] title The title of the section.
+     * \details This only affects the help page and other output formats.
+     */
+    void add_section(std::string const &)
+    {
+        // Not to be implemented in CTD exporter.
+    }
+
+    /*!\brief Adds an help page subsection to the seqan3::argument_parser.
+     * \param[in] title The title of the subsection.
+     * \details This only affects the help page and other output formats.
+     */
+    void add_subsection(std::string const &)
+    {
+        // Not to be implemented in CTD exporter.
+    }
+
+    /*!\brief Adds an help page text line to the seqan3::argument_parser.
+     * \param[in] text The text to print.
+     * \param[in] line_is_paragraph Whether to insert as paragraph
+     *            or just a line (Default: false).
+     * \details
+     * If the line is not a paragraph (false), only one line break is appended, otherwise two line breaks are appended.
+     * This only affects the help page and other output formats.
+     */
+    void add_line(std::string const &, 
+                  bool)
+    {
+        // Not to be implemented in CTD exporter.
+    }
+
+    /*!\brief Adds an help page list item (key-value) to the seqan3::argument_parser.
+     * \param[in] key  The key of the key-value pair of the list item.
+     * \param[in] desc The value of the key-value pair of the list item.
+     */
+    void add_list_item(std::string const &, std::string const &)
+    {
+        // Not to be implemented in CTD exporter.
+    }
+
+private:
+
+    template <typename option_type>
+    bool 
+    value_is_container(option_type const & /* value */) 
+    {
+        return SequenceContainer<option_type> && !std::is_same_v<option_type,
+                                                                 std::string>;
+    }
+
+    // TODO (emanueleparisi) This function is copied from format_parse.hpp, as it is 
```

Comment:
yes, there is the format_base.hpp that you can (and should) inherit from. Then you can move this function into the base class and use it in both.